### DISCUSSION TITLE
EdgeDB 1.0 Alpha 7 "Lalande" is now available for testing

### DISCUSSION THREAD

**Discussion by ambv** [geldata/gel#2053] [Show and tell]
We're happy to announce that the 7th alpha is now ready. Go get it here:
https://edgedb.com/download/

## Improved Migrations

We’ve made a lot of progress in implementing features of the RFC 1000 migrations, although this is still a feature under development. Some of the works can be broadly categorized as overall improvement of the proposed migration DDL and the granularity of the control the user has over these proposed changes. More specifically we’ve made a lot of improvements in migrations that alter or remove things from the schema.

Here’s an example of creating a schema with a type that has a property with a default value:
```edgeql
START MIGRATION TO {
  module default {
    type Foo {
      property val -> str {
        default := 'n/a'
      }
    }
  }
};
```

We use `DESCRIBE CURRENT MIGRATION AS JSON` to see what EdgeDB is proposing. The JSON format makes it easier to potentially integrate this with other tools. For this example it’s worth turning on json output mode for edgedb REPL:

```
\set output-mode json
```

Since proposed statements look OK, we can go ahead and just apply the whole migration.

Now, let’s remove that default, after all the property is optional.

```
START MIGRATION TO {
  module default {
    type Foo {
      property val -> str;
    }
  }
};
```

The proposed statements will `DROP` default for our property, so all seems to be in order and we can apply this migration, too, using `POPULATE MIGRATION` and `COMMIT MIGRATION`.

We’re currently working on a CLI tool for managing migrations more gracefully and without the need for the user to rely on these low-level commands (like `START MIGRATION` or `DESCRIBE CURRENT MIGRATION AS JSON`). The migration tool is going to use these commands behind the scenes, though.

## Command-Line Tools

There's been plenty of quality-of-life additions to the CLI tools. The default user and default database are now simply `edgedb` and no longer named after the system user. A new `--connect-timeout` allows controlling how long to wait for EdgeDB response which is useful for very remote connections. You can now connect to a database instance using a `--dsn`. A new migration-log command views applied migrations. Finally, non-interactive error messages are prefixed by `edgedb error: ...` to quickly spot which tool has errored in scripts.

On the server-side, `edgedb server install` is now friendlier on Linux installations without systemd, allowing foreground execution. When installing server `DEBIAN_FRONTEND` is now noninteractive by default and is overridable. New sub-commands make it easier to maintain an installation:
* `edgedb server logs`;
* `edgedb server info`.

## Bindings

We now have an improved spec for client API (RFC 1004). Rolling out the support for the full spec will be done in the next release, but some implementation work has already started. The `edgedb-python` driver has been upgraded to 0.12.0.

## Correctness improvements

* Update `std::decimal` and `std::bigint` criteria to be sensitive to the presence of `.` so that `123.0e100n` is a decimal, but `123e100n` is a bigint.
* Functions can now return a set of tuples.
* Functions can no longer share a shortname with types in order to avoid name resolution issues.
* Forbid taking implicit cross products with volatile operations to avoid unintuitive behavior.
* Forbid scalar types from having more than one concrete base.
* Forbid partial path expressions in LIMIT/OFFSET clauses.
* Forbid changing cardinality via inheritance.
* Fix cardinality inference on operators.
* Add support for per-database configuration by adding a new CURRENT DATABASE configuration scope.
* Add support for altering enums in order to add new labels.

As always, the full changelog is available here:
https://www.edgedb.com/docs/changelog/1_0_a7

We hope you like this release as we're working hard on bringing you a beta release 🚀

**Comment by vpetrovykh** [comment]
> Since proposed statements look OK, we can go ahead and just apply the whole migration.

> The proposed statements will DROP default for our property, so all seems to be in order and we can apply this migration, too, using POPULATE MIGRATION and COMMIT MIGRATION.

Without the actual output of `DESCRIBE CURRENT MIGRATION` it's not very clear what these two sentences are referencing.

**Reply by 1st1** [reply]
Feel free to edit, Victor.

**Comment by rajeev** [comment]
Wow great!

**Comment by haikyuu** [comment]
Great news. I'm always excited about new releases. I've read the changelog in the PR though 😆 

Good luck with the beta and 1.0
